tests/integration/integration-test.py

```diff
@@ -62,6 +63,110 @@ def test_get_notes_path(self, endpoint='/notes'):
+    
+    def test_get_notes_query_non_existing_student_id(self, endpoint='/notes'):
+        for student_id in self.test_cases['non_existing_student_id']:
+            params = {'studentId': student_id}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            note_schema = utils.get_resource_schema(self, 'NoteResource')
+            utils.check_schema(self, response, note_schema)
+
+        invalid_student_ids = random.sample(range(0, 99999999), 5)
+        # invalid tests returns 400
+        for student_id in invalid_student_ids:
+            params = {'studentId': student_id}
+            response = utils.make_request(self, endpoint, 400, params=params)
+            schema = utils.get_resource_schema(self, 'Error')
+            utils.check_schema(self, response, schema)
+
+    # /notes?studentId=111111111&creatorId
+    def query_creator_id(self, student_id, creator_ids, endpoint='/notes'):
+        for creator_id in creator_ids:
+            params = {'studentId': student_id,'creatorId': creator_id}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            schema = utils.get_resource_schema(self, 'NoteResource')
+            utils.check_schema(self, response, schema)
+            # Validating the creatorId requested is the same creatorId received
+            for resource in response.json()['data']:
+                actual_creator_id = resource['attributes']['creatorId']
+                self.assertEqual(actual_creator_id, creator_id)
+
+    # /notes?studentId=111111111&contextTypes
+    def query_context_types(self, student_id, context_types, endpoint='/notes'):
+        for context_type in context_types:
+            params = {'studentId': student_id,'contextTypes': context_type}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            schema = utils.get_resource_schema(self, 'NoteResource')
+            utils.check_schema(self, response, schema)
+            # Validating the contextTypes requested is the same contextTypes received
+            for resource in response.json()['data']:
+                actual_context_type = resource['attributes']['context']['contextType']
+                self.assertEqual(actual_context_type, context_type)
+
+    # /notes?studentId=111111111&q=
+    def query_string_search(self, student_id, notes, endpoint='/notes'):
+        for note in notes:
+            q = random.choice(note.split(' '))
+            params = {'studentId': student_id, 'q': q}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            schema = utils.get_resource_schema(self, 'NoteResource')
+            utils.check_schema(self, response, schema)
+            # Validating the search word requested exist in the notes received
+            for resource in response.json()['data']:
+                actual_note = resource['attributes']['note']
+                self.assertIn(q, actual_note)
```

Comment:
@ianbrown9475 Ah, I'm not sure too. Please ignore my comment here since I don't think it makes any sense. I guess I was just dizzy or drunk for some reasons :p